subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/DefaultExecutionStateChangeDetector.java

```diff
@@ -122,38 +136,49 @@ public boolean accept(ChangeVisitor visitor) {
 
     private static class DetectedExecutionStateChanges implements ExecutionStateChanges {
         private final InputFileChanges inputFileChanges;
-        private final ChangeContainer allChanges;
-        private final ChangeContainer rebuildTriggeringChanges;
+        private final ImmutableList<String> allChangeMessages;
+        private final boolean rebuildRequired;
```

Comment:
Instead of a field I think we should use two different implementations.

Reply:
Done.